File: Web/src/domain/scores/tunes/oiPidVyshneiu.ts
import type { Score, ScoreEvent } from '../score.js'
import { blow, draw, rest } from '../scoreWriting.js'

const underTheCherryTree: ScoreEvent[] = [
    draw(6, 1), draw(6, 1), draw(6, 1),
    blow(6, 1), draw(5, 2, { vibrato: 0.4 }), blow(6, 1),
    blow(6, 1), blow(6, 1), draw(5, 1),
    blow(5, 2, { vibrato: 0.4 })
]

const theOldManStanding: ScoreEvent[] = [
    draw(4, 1), blow(5, 1), draw(5, 1),
    blow(6, 1), draw(6, 2, { vibrato: 0.4 }), draw(6, 1),
    blow(6, 1), draw(5, 1), blow(5, 1),
    draw(4, 2, { vibrato: 0.4 })
]

export const oiPidVyshneiu: Score = {
    name: 'Oi pid vyshneiu',
    key: 'D',
    position: 'third',
    beatsPerMinute: 96,
    events: [...underTheCherryTree, rest(1), ...theOldManStanding]
}

File: Core/Sources/HarmonicaCore/Entities/Scores/OiPidVyshneiu.swift
public extension Score {
    static let oiPidVyshneiu = Score(
        name: "Oi pid vyshneiu",
        key: .d,
        position: .third,
        beatsPerMinute: 96,
        events: underTheCherryTree + [.rest(beats: 1)] + theOldManStanding
    )

    private static let underTheCherryTree: [ScoreEvent] = [
        .draw(.six, 1), .draw(.six, 1), .draw(.six, 1),
        .blow(.six, 1), .draw(.five, 2, vibrato: 0.4), .blow(.six, 1),
        .blow(.six, 1), .blow(.six, 1), .draw(.five, 1),
        .blow(.five, 2, vibrato: 0.4)
    ]

    private static let theOldManStanding: [ScoreEvent] = [
        .draw(.four, 1), .blow(.five, 1), .draw(.five, 1),
        .blow(.six, 1), .draw(.six, 2, vibrato: 0.4), .draw(.six, 1),
        .blow(.six, 1), .draw(.five, 1), .blow(.five, 1),
        .draw(.four, 2, vibrato: 0.4)
    ]
}
